.devcontainer/devcontainer.json

```diff
@@ -6,7 +6,10 @@
   "features": {
     "ghcr.io/devcontainers-contrib/features/poetry:2": {}
   },
-  // Use 'postCreateCommand' to run commands after the container is created.
+  // This command is kept outside of the postCreateCommand script because we
+  // don't want to include a privileged command in the manual setup process.
+  "onCreateCommand": "sudo /usr/local/py-utils/bin/poetry self add \"poetry-dynamic-versioning[plugin]\"",
```

Comment:
This command is intended to install pdv for only the devcontainer. Because poetry is installed via the devcontainer feature it is installed as root and additional packages can't be installed without elevated permissions. 

Putting bash commands directly in the devcontainer.json itself is the more common pattern. I came up with pattern of referencing a bash script within the devcontainer.json to address the fact that multi-line bash scripts are not well supported by within the json and to enable broader reusability of the setup script (i.e. during manual setup). If a command is a single line and isn't going to be reused in other contexts I prefer to leave it just in the `devcontainer.json`.

Reply:
Makes sense, thanks for explaining this!